client/dashboard/app/router.tsx

```diff
@@ -74,6 +79,31 @@ const siteRoute = createRoute( {
 const siteOverviewRoute = createRoute( {
 	getParentRoute: () => siteRoute,
 	path: '/',
+	loader: async ( { params: { siteSlug } } ) => {
+		// Site usually takes the longest, so kick it off first.
+		const sitePromise = queryClient.ensureQueryData( siteQuery( siteSlug ) );
+		// Kick off all independent promises in parallel.
```

Comment:
I feel like these all are business logic 🤔 we already do this in the component

```
if ( ! site || ! mediaStorage || ! currentPlan || ! primaryDomain || ! engagementStats ) {
		return;
	}
```

so what exactly is the benefit of duplicating it in the `loader` 🤔

Reply:
I'm aware that based on the previous discussion, we do that because TypeScript won't know they are always already non-null. But if we need to do this in the component anyway, then should we not duplicate it in the loader 😄

Reply:
Sorry, I'm still confused. Do we need to ensure that in leaf routes only? I mean why don' we need to wait for `siteQuery` in `siteRoute`'s loeader as well, when the `Site` component which it renders does need the `site` data.

Reply:
You're right, restored it in [Dashboard v2: restore site route loader](https://github.com/Automattic/wp-calypso/pull/103549#top)